src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -2638,5 +2638,270 @@ static void verifyArray(TypeSymbol type)
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_09()
+        {
+            var source = @"
+partial class C
+{
+    public partial ref int M1();
+    public partial int M1() => throw null!; // 1
+
+    public partial int M2();
+    public partial ref int M2() => throw null!; // 2
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,24): error CS8818: Both partial method declarations must return by reference or neither may return by reference.
+                //     public partial int M1() => throw null!; // 1
+                Diagnostic(ErrorCode.ERR_PartialMethodRefReturnDifference, "M1").WithLocation(5, 24),
+                // (8,28): error CS8818: Both partial method declarations must return by reference or neither may return by reference.
+                //     public partial ref int M2() => throw null!; // 2
+                Diagnostic(ErrorCode.ERR_PartialMethodRefReturnDifference, "M2").WithLocation(8, 28));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_10()
+        {
+            var source = @"
+partial class C
+{
+    public partial U M1<U>();
+    public partial U M1<U>() where U : struct => throw null!; // 1
+
+    public partial U M2<U>() where U : struct;
+    public partial U M2<U>() => throw null!; // 2
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,22): error CS0761: Partial method declarations of 'C.M1<U>()' have inconsistent constraints for type parameter 'U'
+                //     public partial U M1<U>() where U : struct => throw null!; // 1
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("C.M1<U>()", "U").WithLocation(5, 22),
+                // (8,22): error CS0761: Partial method declarations of 'C.M2<U>()' have inconsistent constraints for type parameter 'U'
+                //     public partial U M2<U>() => throw null!; // 2
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M2").WithArguments("C.M2<U>()", "U").WithLocation(8, 22));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_11()
+        {
+            var source = @"
+partial class C<T>
+{
+    public partial T M1();
+    public partial T M2();
+}
+
+partial class C<T>
+{
+    public partial T M1() => throw null!;
+    public partial T M2() => throw null!;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
```

Comment:
>} [](start = 8, length = 1)

Please test differences in `dynamic` / `object`, tuple element names, and `nint` / `System.IntPtr`.

Reply:
Added the tests.